zkevm-circuits/src/evm_circuit.rs

```diff
@@ -432,67 +431,52 @@ mod evm_circuit_stats {
         .run();
     }
 
-    /// This function prints to stdout a table with all the implemented states
-    /// and their responsible opcodes with the following stats:
-    /// - height: number of rows in the EVM circuit used by the execution state
-    /// - gas: gas value used for the opcode execution
-    /// - height/gas: ratio between circuit cost and gas cost
+    /// Prints the stats of EVM circuit per execution state.  See
+    /// `print_circuit_stats_by_states` for more details.
     ///
     /// Run with:
-    /// `cargo test -p zkevm-circuits --release get_evm_states_stats --
-    /// --nocapture --ignored`
+    /// `cargo test -p zkevm-circuits --release --all-features
+    /// get_evm_states_stats -- --nocapture --ignored`
     #[ignore]
     #[test]
-    pub fn get_evm_states_stats() {
-        let mut implemented_states = Vec::new();
-        for state in ExecutionState::iter() {
-            let height = state.get_step_height_option();
-            if let Some(h) = height {
-                implemented_states.push((state, h));
-            }
-        }
-
-        let mut stats = Vec::new();
-        for (state, h) in implemented_states {
-            for opcode in state.responsible_opcodes() {
-                let mut code = bytecode! {
-                    PUSH2(0x8000)
-                    PUSH2(0x00)
-                    PUSH2(0x10)
-                    PUSH2(0x20)
-                    PUSH2(0x30)
+    fn get_evm_states_stats() {
+        print_circuit_stats_by_states(
+            |state| {
+                // TODO: Create valid inputs to test all execution states
+                // TODO: Enable CREATE/CREATE2 once they are supported
+                !matches!(
+                    state,
+                    ExecutionState::ErrorInvalidOpcode
+                        | ExecutionState::JUMP
+                        | ExecutionState::JUMPI
+                        | ExecutionState::CREATE
+                        | ExecutionState::CREATE2
+                        | ExecutionState::CALL_OP
```

Comment:
I disabled `CALL_OP` originally because the code used to find the step_index was not working for call-type opcodes (because I was counting from the end).  I've updated the code to find the step_index to be more robust and now it can find all opcodes :)  Also noticed we had a bug in the mock crate that made it not possible to test a call-type opcode to a non-existing account, which I fixed.  So now the only disabled opcodes for the stats are non-implemented ones :)

Reply:
Using pc to find the step looks great! Thanks for adding support for that!